src/bc/model.rs

```diff
@@ -65,6 +213,8 @@ pub(super) struct BcSendInfo {
 #[derive(Debug)]
 pub struct BcContext {
     pub(super) in_bin_mode: HashSet<u32>,
+    pub(super) last_binary_kind: Option<BinaryDataKind>,
+    pub(super) remaining_binary_bytes: usize,
```

Comment:
So everything looks great, except right here there is a subtle issue: the parsing context is actually per-connection, not per-message ID. So with this implementation there would be a problem if there were multiple message IDs with binaryData coming in - they would both overwrite this field.

That's why `in_bin_mode` is a HashSet: I check for the presence/absence of the message ID in that set.

You should do the same thing here, with a HashMap to store your two new fields *per message ID*. If you were more adventurous, you could actually combine `in_bin_mode` with your new HashMap, since their meaning is identical.

Reply:
I haven't look at the msgid but will they be the same for the same set of continuations?

Reply:
Yes, all the video msgids are 3, for example. In practice that's the only binary-enabled message ID I've seen so far, so this hashmap will normally only have one member.

Reply:
Ahh yes I recall looking into that when I first started this branch ok! I am on it

Reply:
Also last_binary_kind and in_bin_mode are not quite the same since I leave it as None if I don't expect a continuation

Reply:
Fair point. I meant more, the presence of your new struct, at all, in the HashMap, should always be the same as the HashSet, making the Set redundant.

Reply:
But since this is not really a performance issue, I am not picky either way.